packages/nitro-protocol/src/signatures.ts

```diff
@@ -15,7 +15,7 @@ export function getStateSignerAddress(signedState: SignedState): string {
   const {channel} = signedState.state;
   const {participants} = channel;
 
-  if (participants.indexOf(recoveredAddress) < 0) {
+  if (participants.map(p => p.toLowerCase()).indexOf(recoveredAddress.toLowerCase()) < 0) {
```

Comment:
Are these conversions still required? It seems like we should know that the addresses are all lower case by this point?

Reply:
Good question. In this specific case, `recoveredAddress` will be a checksum as that address is [derived using ethers ](https://github.com/statechannels/statechannels/blob/3ac43c32923f09ab9a1c21efa4faa7243cc2b957/packages/nitro-protocol/src/signatures.ts#L14). The question, then, is should we assume that the participants array contains lower case addresses?

This code is part of the `nitro-protocol` package. We currently do **not** enforce/assume lowercase addresses or bytes in nitro protocol. We only do so in `server-wallet` and `wallet-core` packages.  Why don't we not assume lower case addresses in `nitro-protocol`? My thinking was that I did not want to create a breaking change in `nitro-protocol`. I was primarily thinking about not breaking`xstate-wallet`.

However, `xstate-wallet` depends on `wallet-core`. So the breaking changes in `wallet-core` required pinning `xstate-wallet` to a version of `wallet-core` before this PR. So perhaps we should just introduce the breaking change into `nitro-protocol`?

Reply:
After giving this more thought: if we are going to enforce lower cases in `nitro-protocol`, should `Address` and `Destination` intersection types be defined in the `nitro-protocol` package?

Reply:
During the discussion in standup, the conclusion was to enforce address and bytes cases in nitro protocol. In the interest of reducing the changeset of this PR, leaving this work for future PRs: https://github.com/statechannels/statechannels/issues/2859